debezium/scripts/demo.py

```diff
@@ -36,133 +38,133 @@ def make_cdc_table(table_name:str):
+    key_spec = KeyValueSpec.IGNORE,
+    value_spec = ck.json_spec([ ('user_id', dh.int_),
                       ('url', dh.string),
                       ('channel', dh.string),
-                      ('received_at', dh.datetime) ]),
-    table_type = 'append'
+                      ('received_at', dh.DateTime) ]),
+    table_type = TableType.Append
 )
 
 pageviews_stg = pageviews \
-    .updateView(
+    .update_view([
         'url_path = url.split(`/`)',
         'pageview_type = url_path[1]',
         'target_id = Long.parseLong(url_path[2])'
-    ).dropColumns('url_path')
-
-purchases_by_item = purchases.aggBy(
-    as_list([
-        agg.AggSum('revenue = purchase_price'),
-        agg.AggCount('orders'),
-        agg.AggSum('items_sold = quantity')
-    ]),
+    ]).drop_columns('url_path')
+
+purchases_by_item = purchases.agg_by(
+    [
+        agg.sum_(['revenue = purchase_price']),
+        agg.count_('orders'),
+        agg.sum_(['items_sold = quantity'])
+    ],
     'item_id'
 )
 
 pageviews_by_item = pageviews_stg \
-    .where('pageview_type = `products`') \
-    .countBy('pageviews', 'item_id = target_id')
+    .where(['pageview_type = `products`']) \
+    .count_by('pageviews', ['item_id = target_id'])
 
 item_summary = items \
-    .view('item_id = id', 'name', 'category') \
-    .naturalJoin(purchases_by_item, 'item_id') \
-    .naturalJoin(pageviews_by_item, 'item_id') \
-    .dropColumns('item_id') \
-    .moveColumnsDown('revenue', 'pageviews') \
-    .updateView('conversion_rate = orders / (double) pageviews')
+    .view(['item_id = id', 'name', 'category']) \
+    .natural_join(purchases_by_item, on = ['item_id']) \
+    .natural_join(pageviews_by_item, on = ['item_id']) \
+    .drop_columns('item_id') \
+    .move_columns_down(['revenue', 'pageviews']) \
+    .update_view(['conversion_rate = orders / (double) pageviews'])
 
 # These two 'top_*' tables match the 'Business Intelligence: Metabase' / dashboard
 # part of the original example.
 top_viewed_items = item_summary \
-    .sortDescending('pageviews') \
-    .head(20)
+        .sort_descending('pageviews') \
+        .head(20)
 
 top_converting_items = item_summary \
-    .sortDescending('conversion_rate') \
+    .sort_descending('conversion_rate') \
     .head(20)
 
 minute_in_nanos = 60 * 1000 * 1000 * 1000
 
 profile_views_per_minute_last_10 = \
-    wck.addTimeWindow(
-        pageviews_stg.where('pageview_type = `profiles`'),
-        'received_at',
-        10*minute_in_nanos,
-        'in_last_10min'
+    time_window(
+        pageviews_stg.where(['pageview_type = `profiles`']),
+        ts_col='received_at',
+        window=10*minute_in_nanos,
+        bool_col='in_last_10min'
     ).where(
-        'in_last_10min = true'
-    ).updateView(
-        'received_at_minute = lowerBin(received_at, minute_in_nanos)'
+        ['in_last_10min = true']
+    ).update_view(
+        ['received_at_minute = (DateTime)lower_bin(received_at, minute_in_nanos)']
```

Comment:
I am confused as for why we would need a cast here.  Can we figure out what the types are?

Reply:
resolved that parser would be useful for situations like this. In the mean time using lowerBin.